src/memo_ingredient_indices.rs

```diff
@@ -49,66 +29,64 @@ impl IngredientIndices {
         &self,
         mut memo_index: impl FnMut(IngredientIndex) -> MemoIngredientIndex,
     ) -> MemoIngredientIndices {
-        let mut memo_ingredient_indices = [(
-            IngredientIndex::from((u32::MAX - 1) as usize),
+        let mut indices = Vec::new();
+        indices.resize(
+            self.indices.last().unwrap().as_usize() + 1,
```

Comment:
mind replacing this with an `.expect()`?

Reply:
Uuh right I forgot that I need to handle this actually